polygon/client.go

```
@@ -405,6 +400,21 @@ func (ec *Client) getBlock(
 	return types.NewBlockWithHeader(&head).WithBody(txs, uncles), loadedTxs, nil
 }
 
+// effectiveGasPrice returns the price of gas charged to this transaction to be included in the
+// block.
+func effectiveGasPrice(tx *EthTypes.Transaction, baseFee *big.Int) *big.Int {
+	if tx.Type() != eip1559TxType {
+		return tx.GasPrice()
+	}
+	// For EIP-1559 the gas price is determined by the base fee & miner tip instead
+	// of the tx-specified gas price.
+	tip, err := tx.EffectiveGasTip(baseFee)
+	if err != nil {
```

Do we want to propagate this error? Or maybe not since it looks like there's only 1 potential error type that would be returned: https://github.com/ethereum/go-ethereum/blob/045e90c8971cddbabeac0abd54240cf02bc1d94d/core/types/transaction.go#L333

---

Yeah I was taking a shortcut here since the error shouldn't actually happen .. but if it does it would be good to know the error (who knows there might be more than one in the future).  Updated to propagate the error.